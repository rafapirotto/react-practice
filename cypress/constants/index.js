export const HOME_URL = 'http://localhost:3001/';
export const MY_LIST_URL = `${HOME_URL}my-list`;
export const TRAILERS_ROUTE = `${HOME_URL}trailers/`;
