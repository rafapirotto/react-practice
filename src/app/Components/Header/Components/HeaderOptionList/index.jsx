import React from 'react';

import HeaderOption from '../HeaderOption';

import search from '../../assets/search.png';
import userImage from '../../assets/user-image.png';
import arrow from '../../assets/arrow.png';

import styles from './styles/HeaderOptionList.module.css';

const HeaderOptionList = () => (
  <div className="flex items-center flex-grow justify-end">
    <HeaderOption src={search} alt="search" styles={styles.search} />
    <HeaderOption src={userImage} alt="user" styles={styles.userImage} />
    <HeaderOption src={arrow} alt="arrow" styles={styles.arrow} />
  </div>
);

export default HeaderOptionList;
