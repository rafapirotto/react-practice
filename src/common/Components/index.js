import Error from './Error';
import Spinner from './Spinner';
import StarRating from './StarRating';
import ProviderRouterWrapper from './ProviderRouterWrapper';
import PrivateRoute from './PrivateRoute';
import PublicRoute from './PublicRoute';

export { Error, Spinner, StarRating, ProviderRouterWrapper, PrivateRoute, PublicRoute };
