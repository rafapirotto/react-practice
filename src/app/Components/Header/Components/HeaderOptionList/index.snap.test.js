import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';

import HeaderOptionList from '.';
import { SUCCESS } from '../../../../../common/constants';

it('Matches the snapshot', () => {
  const middlewares = [thunk];
  const mockStore = configureStore(middlewares);
  const initialState = {
    userProfile: {
      state: SUCCESS,
      content: { photo_path: 'https://randomuser.me/api/portraits/med/men/75.jpg' },
    },
  };
  const store = mockStore(initialState);

  const component = renderer.create(
    <Router>
      <Provider store={store}>
        <HeaderOptionList />
      </Provider>
    </Router>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
