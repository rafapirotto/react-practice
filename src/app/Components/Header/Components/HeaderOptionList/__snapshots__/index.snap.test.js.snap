// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshot 1`] = `
<div
  className="flex items-center flex-grow justify-end"
>
  <a
    href="/"
    onClick={[Function]}
  >
    <img
      alt="search"
      className="mx-3 search"
      src="search.png"
    />
  </a>
  <a
    href="/"
    onClick={[Function]}
  >
    <img
      alt="user"
      className="mx-3 userImage"
      src="user-image.png"
    />
  </a>
  <a
    href="/"
    onClick={[Function]}
  >
    <img
      alt="arrow"
      className="mx-3 arrow"
      src="arrow.png"
    />
  </a>
</div>
`;
